File: chapter1/salesitemservice/src/main/java/com/example/salesitemservice/SalesItemService.java
package com.example.salesitemservice;


public interface SalesItemService {
  SalesItem createSalesItem(InputSalesItem inputSalesItem);

  SalesItem getSalesItemById(Long id);

  Iterable<SalesItem> getSalesItemsByUserAccountId(Long userAccountId);

  Iterable<SalesItem> getSalesItems();

  void updateSalesItem(Long id, InputSalesItem inputSalesItem);

  void deleteSalesItemById(Long id);

  void deleteSalesItems();
}

File: chapter6/salesitemservice_graphql/src/services/SalesItemService.ts
import InputSalesItem from '../dtos/InputSalesItem';
import OutputSalesItem from '../dtos/OutputSalesItem';

export default interface SalesItemService {
  createSalesItem(inputSalesItem: InputSalesItem): Promise<OutputSalesItem>;

  getSalesItems(): Promise<OutputSalesItem[]>;

  getSalesItem(id: string): Promise<OutputSalesItem>;

  updateSalesItem(id: string, inputSalesItem: InputSalesItem): Promise<void>;

  deleteSalesItem(id: string): Promise<void>;
}
